### PR TITLE
Add mixed CPU/GPU exawind-driver capability for Eagle and Summit

### PR DESCRIPTION
**Pull request by jrood-nrel**
This also:
 1. Generalizes the darwin `compilers.yaml`
 2. Updates the `install-exawind.sh` script to use `create_machine_spack_environment.py`
 3. Moves the stage directory by default to a spack-manager stage directory
 4. Moves the cache directory by default to a spack-manager cache directory
 5. Updates the Spack submodule
 6. Includes the specific newer `ld` required on Eagle in `compilers.yaml`

### CODE REVIEW

**Comment by jrood-nrel** [comment]
@psakievich Do you want to update the script to avoid blank yaml files? Then I can just have Eagle build clingo and we can keep the concretizer as clingo.

File: scripts/install-exawind.sh [hunk 1]
```diff
@@ -9,19 +12,34 @@ set -e
 
 MYPWD=${PWD}
 
+cmd "export TMPDIR=${MYPWD}/../tmp"
+cmd "export SPACK_MANAGER=${MYPWD}/.."
+cmd "source ${SPACK_MANAGER}/start.sh"
+
 if [ "${SPACK_MANAGER_MACHINE}" == 'summit' ]; then
   cmd "module unload xl"
   cmd "module load python/3.7.7"
```
Comment (psakievich) [inline scripts/install-exawind.sh:21]: Just a side note, but have you tried putting these in the `compilers.yaml` for eagle?

**Comment by psakievich** [comment]
> @psakievich Do you want to update the script to avoid blank yaml files? Then I can just have Eagle build clingo and we can keep the concretizer as clingo.

@jrood-nrel yeah i'll update that tonight.

File: scripts/install-exawind.sh (hunk 1 above)
Comment (jrood-nrel) [reply scripts/install-exawind.sh:21]: I guess I should put binutils in there.

Comment (jrood-nrel) [reply scripts/install-exawind.sh:21]: I added this.

**Review by psakievich — APPROVED** [review]

**Pull request merged**